File: gravitee-management-api-model/src/main/java/io/gravitee/management/model/api/NewApiEntity.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.management.model.api;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.Set;

/**
 * @author David BRASSELY (david.brassely at graviteesource.com)
 * @author Nicolas GERAUD (nicolas.geraud at graviteesource.com)
 * @author GraviteeSource Team
 */
public class NewApiEntity {

    @NotNull
    @NotEmpty(message = "Api's name must not be empty")
    private String name;

    @NotNull
    private String version;

    @NotNull
    private String description;

    @NotNull
    private String contextPath;

    @NotNull
    private String endpoint;

    private Set<String> groups;

    private List<String> paths;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getContextPath() {
        return contextPath;
    }

    public void setContextPath(String contextPath) {
        this.contextPath = contextPath;
    }

    public String getEndpoint() {
        return endpoint;
    }

    public void setEndpoint(String endpoint) {
        this.endpoint = endpoint;
    }

    public List<String> getPaths() {
        return paths;
    }

    public void setPaths(List<String> paths) {
        this.paths = paths;
    }

    public Set<String> getGroups() {
        return groups;
    }

    public void setGroups(Set<String> groups) {
        this.groups = groups;
    }

    @Override
    public String toString() {
        return "NewApiEntity{" +
                "name='" + name + '\'' +
                ", version='" + version + '\'' +
                ", description='" + description + '\'' +
                ", contextPath='" + contextPath + '\'' +
                ", endpoint='" + endpoint + '\'' +
                ", groups='" + groups + '\'' +
                '}';
    }
}

File: gravitee-apim-console-webui/src/entities/api-v4/NewApiEntity.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ApiType } from './ApiType';
import { EndpointGroup } from './EndpointGroup';
import { Flow } from './Flow';
import { FlowMode } from './FlowMode';
import { Listener } from './Listener';

export interface NewApiEntity {
  name?: string;
  apiVersion?: string;
  definitionVersion?: '4.0.0';
  type?: ApiType;
  description?: string;
  tags?: string[];
  listeners?: Listener[];
  endpointGroups?: EndpointGroup[];
  flowMode?: FlowMode;
  flows?: Flow[];
  groups?: string[];
}
